scalyr_agent/copying_manager.py

```diff
@@ -1029,8 +1033,17 @@ def __remove_logs_scheduled_for_deletion( self ):
                         self.__logs_pending_removal.pop( path, None )
                     finally:
                         self.__lock.release()
+                    removed_paths[path] = True
                     log.warn( "Log scheduled for removal is not being monitored: %s" % path )
 
+        # update log matchers
+        self.__lock.acquire()
+        try:
+            self.__log_matchers[:] = [m for m in self.__log_matchers if m.log_path not in removed_paths]
```

Comment:
The call to `remove_paths` updates the `__log_matchers` variable already, and it does it only if the number of paths has reached zero.  Should we really be doing the update here?  And why don't we have to do the same checks?